src/compiler/utilities.ts

```diff
@@ -2685,64 +2602,55 @@ namespace ts {
         action: (emitFileNames: EmitFileNames, sourceFiles: SourceFile[], isBundledEmit: boolean, emitOnlyDtsFiles: boolean) => void,
         targetSourceFile?: SourceFile,
         emitOnlyDtsFiles?: boolean) {
+        forEachEmittedFile(host, getSourceFilesToEmit(host, targetSourceFile), action, emitOnlyDtsFiles);
+    }
+
+    /**
+     * Iterates over each source file to emit.
+     */
+    export function forEachEmittedFile(host: EmitHost, sourceFiles: SourceFile[],

```

Comment:
I would suggest combining this with ```forEachExpectedEmitFile``` instead (eg. taking sourceFiles list as optional input) so as to get rid of duplicate logic here.